smartmin/views.py

```diff
@@ -419,21 +414,38 @@ class SmartTemplateView(SmartView, TemplateView):
     pass
 
 
-class SmartReadView(SmartView, DetailView):
+def derive_single_object_url_pattern(slug_url_kwarg, path, action):
+    """
+    Utility function called by class methods for single object views
+    """
+    if slug_url_kwarg:
+        return r'^%s/%s/(?P<%s>[^/]+)/$' % (path, action, slug_url_kwarg)
+    else:
+        return r'^%s/%s/(?P<pk>\d+)/$' % (path, action)
```

Comment:
This allows for building the correct URL based on either a PK or a slug field (think uuid). Since in Smartmin land this needs to be a classmethod, we can't use inheritance, so instead it is a utility method.

Previously only the Read view supported this, but now all single object views support it.